### PR TITLE
Disallow VIEW in OUTPUT INTO clause

### PR DESCRIPTION
**Pull request by vikashprajapati-cell**
### Description

This PR adds a fix to disallow `VIEW` in OUTPUT INTO clause. So, when query of type`UPDATE table1 SET c1 = 23 OUTPUT INSERTED.c2, 22 INTO view1` executes on babelfish, it executes without throwing any error to client. Ideally it should throw error. So, to fix this, the `relkind`(Type of relation)  is fetched using `relid`(Relation id) and compared this relation type to find the type of relation in OUTPUT INTO clause. 
This PR contains the changes for VIEW relation only. And for common table expression(CTE), babel does not support the syntax as of now. 

Task: BABEL-2144
Authored-by: Vikash Prajapati [vikasprj@amazon.com](mailto:vikasprj@amazon.com)
Signed-off-by: Vikash Prajapati [vikasprj@amazon.com](mailto:vikasprj@amazon.com)

### Issues Resolved

BABEL-2144

### Test Scenarios Covered ###
* **Use case based -**
Added

* **Boundary conditions -** **NA**

* **Arbitrary inputs -** **NA**

* **Negative test cases -** **NA**

* **Minor version upgrade tests -** **NA**

* **Major version upgrade tests -** **NA**

* **Performance tests -** **NA**

* **Tooling impact -** **NA**

* **Client tests -** **NA**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 1]
```diff
@@ -2324,6 +2324,14 @@ modify_insert_stmt(InsertStmt *stmt, Oid relid)
 	HeapTuple	tuple;
 	List	   *insert_col_list = NIL,
 			   *temp_col_list;
+	char		relkind = get_rel_relkind(relid);
+
+	if(relkind == RELKIND_VIEW || relkind == RELKIND_MATVIEW)
+	{
+		ereport(ERROR,
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/src/hooks.c:2346]: We should map this error code to TSQL error code.

File: test/JDBC/input/BABEL-2144-vu-cleanup.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+drop view babel_2144_vu_prepare_v1;
+go
+drop table babel_2144_vu_prepare_t1;
+go
+drop table babel_2144_vu_prepare_t2;
+go
```
Comment (rishabhtanwar29) [inline test/JDBC/input/BABEL-2144-vu-cleanup.sql]: Nit: Add a new line at the end of each new file.

**Comment by rishabhtanwar29** [comment]
Suggestion: Title should be: `Disallow VIEW in OUTPUT INTO clause`

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 1 above)
Comment (vikashprajapati-cell) [reply contrib/babelfishpg_tsql/src/hooks.c:2346]: Done

File: test/JDBC/input/BABEL-2144-vu-cleanup.sql (hunk 2 above)
Comment (vikashprajapati-cell) [reply test/JDBC/input/BABEL-2144-vu-cleanup.sql]: Done!

**Review by rishabhtanwar29 — APPROVED** [review]
LGTM

**Review by ahmed-shameem — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**